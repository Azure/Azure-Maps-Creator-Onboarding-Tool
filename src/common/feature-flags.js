const flags = {
  onboardingEnabled: false,
};

export default flags;