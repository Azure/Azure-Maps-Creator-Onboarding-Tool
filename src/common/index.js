import { PATHS, ROUTE_NAME_BY_PATH } from './constants';

export { PATHS, ROUTE_NAME_BY_PATH };