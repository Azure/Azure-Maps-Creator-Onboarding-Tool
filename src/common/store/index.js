export { conversionSteps, useConversionStore } from './conversion.store';
export { useDissolvedExterior, useGeometryStore } from './geometry.store';
export { useLayersStore } from './layers.store';
export { useLevelsStore } from './levels.store';
export {
  progressBarStepsByKey,
  useCompletedSteps,
  useProgressBarSteps,
  useProgressBarStore,
} from './progress-bar-steps';
export { resetStores } from './reset';
export { LRO_STATUS, useResponseStore } from './response.store';
export { useReviewManifestJson, useReviewManifestStore } from './review-manifest.store';
export { getDomain, useUserStore } from './user.store';
