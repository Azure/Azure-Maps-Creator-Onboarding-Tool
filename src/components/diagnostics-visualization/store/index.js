import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

import rootReducer from '../reducers';

// Middlewares go here
const middleware = {};

const configureStore = (initialState) => createStore(
  rootReducer,
  initialState,
  // Redux Dev Tools Middleware
  composeWithDevTools(applyMiddleware(...middleware)),
);

// pass an optional param to rehydrate state on app start
const store = configureStore();

// export store singleton instance
export default store;
