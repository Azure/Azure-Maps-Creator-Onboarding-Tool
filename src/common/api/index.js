// The following API is designed to be used only by the onboarding tool and is not supported for any other use.
import { getEnvs } from 'common/functions';
import { useUserStore } from '../store/user.store';

export const uploadFile = (file) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  const url = `${getEnvs()[geography].URL}/manifest?api-version=2.0&subscription-key=${subscriptionKey}`;

  return fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/zip',
    },
    body: file,
  });
};

export const fetchFromLocation = (location) => {
  const { subscriptionKey } = useUserStore.getState();
  return fetch(`${location}&subscription-key=${subscriptionKey}`, {
    method: 'GET',
  });
};

export const deleteFromLocation = (fetchUrl) => {
  const { subscriptionKey } = useUserStore.getState();
  fetch(`${fetchUrl}&subscription-key=${subscriptionKey}`, {
    method: 'DELETE',
  });
};

export const fetchAddress = (address) => {
  const { geography, subscriptionKey } = useUserStore.getState();
  return fetch(`${getEnvs()[geography].URL}/search/address/json?subscription-key=${subscriptionKey}&api-version=1.0&query=${address}&limit=1`)
    .then((res) => res.json());
};