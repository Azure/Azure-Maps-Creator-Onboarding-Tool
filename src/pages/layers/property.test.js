import { fireEvent, render, screen } from '@testing-library/react';

import Property from './property';
import { useLayersStore } from 'common/store';

describe('Property component', () => {
  beforeEach(() => {
    useLayersStore.setState({
      textLayerNames: ['textLayer1', 'textLayer2', 'textLayer3'],
      layers: [
        {
          id: 'id12',
          value: ['polygonLayer21'],
          name: 'base layer',
          isDraft: false,
          props: [
            {
              id: 'id11',
              name: 'Propp',
              value: ['textLayer2', 'textLayer3'],
              isDraft: false,
            },
          ],
        },
      ],
    });
  });

  it('should render component', () => {
    const view = render(
      <Property parentId={'id12'} id={'id11'} name="Propp" value={['textLayer2', 'textLayer3']} isDraft={false} />
    );
    expect(view).toMatchSnapshot();
  });

  it('should render options correctly', () => {
    const view = render(
      <Property parentId={'id12'} id={'id11'} name="Propp" value={['textLayer2', 'textLayer3']} isDraft={false} />
    );
    const dropdown = screen.getByRole('combobox');
    fireEvent.click(dropdown);
    expect(view).toMatchSnapshot();
  });

  it('should call updateProperty when changing selected options', () => {
    const state = useLayersStore.getState();
    const spy = jest.spyOn(state, 'updateProperty');
    render(
      <Property parentId={'id12'} id={'id11'} name="Propp" value={['textLayer2', 'textLayer3']} isDraft={false} />
    );
    const dropdown = screen.getByRole('combobox');
    fireEvent.click(dropdown);
    const optionsList = screen.getAllByRole('menuitemcheckbox');
    fireEvent.click(optionsList[0]);
    expect(spy).toHaveBeenCalledWith('id12', 'id11', { value: ['textLayer2', 'textLayer3', 'textLayer1'] });
  });

  it('should delete prop when delete button was clicked', () => {
    const state = useLayersStore.getState();
    const spy = jest.spyOn(state, 'deleteProperty');
    render(
      <Property parentId={'id12'} id={'id11'} name="Propp" value={['textLayer2', 'textLayer3']} isDraft={false} />
    );
    const deleteBtn = screen.getByLabelText('delete.property');
    fireEvent.click(deleteBtn);
    expect(spy).toHaveBeenCalled();
  });
});
