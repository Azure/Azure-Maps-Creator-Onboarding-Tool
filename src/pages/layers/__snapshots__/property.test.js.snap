// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property component should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: calc(100% - 2rem);
  margin-top: 0.5rem;
  padding-left: 2rem;
}

.emotion-1 {
  width: calc(50% - 2.5rem);
  margin-right: 1.25rem;
}

.emotion-2 {
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  width: calc(50% - 1.5rem);
  width: calc(50% - 0.5rem);
}

.emotion-3 {
  font-size: 0.813rem;
  color: black;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.emotion-4 {
  width: 1.5rem;
  margin-left: 0.25rem;
}

.emotion-5 {
  color: black;
  font-size: 0.813rem;
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-5:hover {
  color: black;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="ms-TextField emotion-1 root-110"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-111"
            >
              <input
                aria-invalid="false"
                class="ms-TextField-field field-112"
                id="TextField0"
                placeholder=""
                type="text"
                value="Propp"
              />
            </div>
          </div>
        </div>
        <div
          class="fui-Dropdown emotion-2 ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
          style="border-radius: 2px; border: 1px solid #605e5c; font-size: 0.813rem; height: 1.5rem; line-height: 1.5rem;"
        >
          <button
            aria-expanded="false"
            class="fui-Dropdown__button ___1jkuyn2_1ro7mo8 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl fxc4j92 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fy9rknc figsok6 fwrc4pm f1khb0e9 fdw0yi8 f1jnq6q7 f1xile11"
            role="combobox"
            type="button"
          >
            <div
              class="emotion-3"
            >
              textLayer2, textLayer3
            </div>
            <span
              class="fui-Dropdown__expandIcon ___wu62wz0_l0el1i0 f1ewtqcl fxkbij4 ftgm304 f4ybsrx f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
            >
              <svg
                aria-hidden="true"
                class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                fill="currentColor"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
        </div>
        <div
          class="emotion-4"
        >
          <button
            aria-label="delete.property"
            class="ms-Button ms-Button--icon emotion-5 root-121"
            data-is-focusable="true"
            data-testid="delete-icon"
            title="delete.property"
            type="button"
          >
            <span
              class="ms-Button-flexContainer flexContainer-122"
              data-automationid="splitbuttonprimary"
            >
              <i
                aria-hidden="true"
                class="ms-Icon root-105 emotion-6 ms-Button-icon icon-124"
                data-icon-name="StatusCircleErrorX"
                style="font-family: \\"FabricMDL2Icons-9\\";"
              >
                
              </i>
            </span>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: calc(100% - 2rem);
  margin-top: 0.5rem;
  padding-left: 2rem;
}

.emotion-1 {
  width: calc(50% - 2.5rem);
  margin-right: 1.25rem;
}

.emotion-2 {
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  width: calc(50% - 1.5rem);
  width: calc(50% - 0.5rem);
}

.emotion-3 {
  font-size: 0.813rem;
  color: black;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.emotion-4 {
  width: 1.5rem;
  margin-left: 0.25rem;
}

.emotion-5 {
  color: black;
  font-size: 0.813rem;
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-5:hover {
  color: black;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="ms-TextField emotion-1 root-110"
      >
        <div
          class="ms-TextField-wrapper"
        >
          <div
            class="ms-TextField-fieldGroup fieldGroup-111"
          >
            <input
              aria-invalid="false"
              class="ms-TextField-field field-112"
              id="TextField0"
              placeholder=""
              type="text"
              value="Propp"
            />
          </div>
        </div>
      </div>
      <div
        class="fui-Dropdown emotion-2 ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
        style="border-radius: 2px; border: 1px solid #605e5c; font-size: 0.813rem; height: 1.5rem; line-height: 1.5rem;"
      >
        <button
          aria-expanded="false"
          class="fui-Dropdown__button ___1jkuyn2_1ro7mo8 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl fxc4j92 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fy9rknc figsok6 fwrc4pm f1khb0e9 fdw0yi8 f1jnq6q7 f1xile11"
          role="combobox"
          type="button"
        >
          <div
            class="emotion-3"
          >
            textLayer2, textLayer3
          </div>
          <span
            class="fui-Dropdown__expandIcon ___wu62wz0_l0el1i0 f1ewtqcl fxkbij4 ftgm304 f4ybsrx f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        class="emotion-4"
      >
        <button
          aria-label="delete.property"
          class="ms-Button ms-Button--icon emotion-5 root-121"
          data-is-focusable="true"
          data-testid="delete-icon"
          title="delete.property"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-122"
            data-automationid="splitbuttonprimary"
          >
            <i
              aria-hidden="true"
              class="ms-Icon root-105 emotion-6 ms-Button-icon icon-124"
              data-icon-name="StatusCircleErrorX"
              style="font-family: \\"FabricMDL2Icons-9\\";"
            >
              
            </i>
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Property component should render options correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: calc(100% - 2rem);
  margin-top: 0.5rem;
  padding-left: 2rem;
}

.emotion-1 {
  width: calc(50% - 2.5rem);
  margin-right: 1.25rem;
}

.emotion-2 {
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  width: calc(50% - 1.5rem);
  width: calc(50% - 0.5rem);
}

.emotion-3 {
  font-size: 0.813rem;
  color: black;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.emotion-4 {
  width: 1.5rem;
  margin-left: 0.25rem;
}

.emotion-5 {
  color: black;
  font-size: 0.813rem;
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-5:hover {
  color: black;
}

<body>
    <div>
      <div
        class="emotion-0"
      >
        <div
          class="ms-TextField emotion-1 root-110"
        >
          <div
            class="ms-TextField-wrapper"
          >
            <div
              class="ms-TextField-fieldGroup fieldGroup-111"
            >
              <input
                aria-invalid="false"
                class="ms-TextField-field field-112"
                id="TextField8"
                placeholder=""
                type="text"
                value="Propp"
              />
            </div>
          </div>
        </div>
        <div
          aria-owns="fluent-listbox2"
          class="fui-Dropdown emotion-2 ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
          style="border-radius: 2px; border: 1px solid #605e5c; font-size: 0.813rem; height: 1.5rem; line-height: 1.5rem;"
        >
          <button
            aria-activedescendant="fluent-option3"
            aria-expanded="true"
            class="fui-Dropdown__button ___1jkuyn2_1ro7mo8 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl fxc4j92 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fy9rknc figsok6 fwrc4pm f1khb0e9 fdw0yi8 f1jnq6q7 f1xile11"
            role="combobox"
            type="button"
          >
            <div
              class="emotion-3"
            >
              textLayer2, textLayer3
            </div>
            <span
              class="fui-Dropdown__expandIcon ___wu62wz0_l0el1i0 f1ewtqcl fxkbij4 ftgm304 f4ybsrx f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
            >
              <svg
                aria-hidden="true"
                class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
                fill="currentColor"
                height="1em"
                viewBox="0 0 20 20"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                  fill="currentColor"
                />
              </svg>
            </span>
          </button>
          <span
            hidden=""
          />
        </div>
        <div
          class="emotion-4"
        >
          <button
            aria-label="delete.property"
            class="ms-Button ms-Button--icon emotion-5 root-121"
            data-is-focusable="true"
            data-testid="delete-icon"
            title="delete.property"
            type="button"
          >
            <span
              class="ms-Button-flexContainer flexContainer-122"
              data-automationid="splitbuttonprimary"
            >
              <i
                aria-hidden="true"
                class="ms-Icon root-105 emotion-6 ms-Button-icon icon-124"
                data-icon-name="StatusCircleErrorX"
                style="font-family: \\"FabricMDL2Icons-9\\";"
              >
                
              </i>
            </span>
          </button>
        </div>
      </div>
    </div>
    <div
      class="___1uxlrft_djprso0 f1euv43f f15twtuk f1vgc2s3 f1e31b4d f494woh"
      dir="ltr"
    >
      <div
        aria-multiselectable="true"
        class="fui-Listbox fui-Dropdown__listbox ___1te21m2_r0cuxf0 fxugw4r f1ewtqcl f22iagw f1vx9l62 f3hsy1e f5zp4f fpvhumw f1yog68k f13sgyd8 f1x4af0m f7x41pl fd55psn fruq291 fiut8dr f1hg901r f1aa9q02 f16jpd5f f1jar5jt fyu767a fmmk62d"
        id="fluent-listbox2"
        role="menu"
        style="position: fixed; left: 0px; top: 0px; margin: 0px; width: 0px;"
      >
        <div
          aria-checked="false"
          class="fui-Option ___1u9whar_q1puvv0 f122n59 f1aa9q02 f16jpd5f f1jar5jt fyu767a f19n0e5 f1ufnopg f1k6fduh f22iagw fk6fouc fkhj508 f1i3iumi fp2oml8 f1vdfbxk f1tdddsa f1f5gg8d f10pi13n f1knas48 fb40n2d"
          id="fluent-option3"
          role="menuitemcheckbox"
          style="font-size: 0.813rem; color: black;"
        >
          <span
            aria-hidden="true"
            class="fui-Option__checkIcon ___12nzoej_6q04f40 f1ugzwwg f18b9hdq f1xk557c f1022m68 fvc9v3g f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fq0vr37 f1byw159 f1c1zstj f11cr0be f1g3puop fi2rrw2 f1rstyi9 f1s4nn1u f1ewtqcl f22iagw f122n59 f4d9j23 f1w9h62z fd461yt fjw5fx7"
          />
          textLayer1
        </div>
        <div
          aria-checked="true"
          class="fui-Option ___1u9whar_q1puvv0 f122n59 f1aa9q02 f16jpd5f f1jar5jt fyu767a f19n0e5 f1ufnopg f1k6fduh f22iagw fk6fouc fkhj508 f1i3iumi fp2oml8 f1vdfbxk f1tdddsa f1f5gg8d f10pi13n f1knas48 fb40n2d"
          id="fluent-option4"
          role="menuitemcheckbox"
          style="font-size: 0.813rem; color: black;"
        >
          <span
            aria-hidden="true"
            class="fui-Option__checkIcon ___1sfj1o1_17ody2f f1ugzwwg f18b9hdq f1xk557c f1022m68 fvc9v3g f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z f3xi7mh ftovhe4 f68vbr6 f1wczvin f1g3puop fi2rrw2 f1rstyi9 f1s4nn1u f1ewtqcl f22iagw f122n59 f4d9j23 f1w9h62z fd461yt fjw5fx7 ftywsgz fqpbvvt"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="12"
              viewBox="0 0 12 12"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.76 3.2c.3.29.32.76.04 1.06l-4.25 4.5a.75.75 0 0 1-1.08.02L2.22 6.53a.75.75 0 0 1 1.06-1.06l1.7 1.7L8.7 3.24a.75.75 0 0 1 1.06-.04Z"
                fill="currentColor"
              />
            </svg>
          </span>
          textLayer2
        </div>
        <div
          aria-checked="true"
          class="fui-Option ___1u9whar_q1puvv0 f122n59 f1aa9q02 f16jpd5f f1jar5jt fyu767a f19n0e5 f1ufnopg f1k6fduh f22iagw fk6fouc fkhj508 f1i3iumi fp2oml8 f1vdfbxk f1tdddsa f1f5gg8d f10pi13n f1knas48 fb40n2d"
          id="fluent-option5"
          role="menuitemcheckbox"
          style="font-size: 0.813rem; color: black;"
        >
          <span
            aria-hidden="true"
            class="fui-Option__checkIcon ___1sfj1o1_17ody2f f1ugzwwg f18b9hdq f1xk557c f1022m68 fvc9v3g f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z f3xi7mh ftovhe4 f68vbr6 f1wczvin f1g3puop fi2rrw2 f1rstyi9 f1s4nn1u f1ewtqcl f22iagw f122n59 f4d9j23 f1w9h62z fd461yt fjw5fx7 ftywsgz fqpbvvt"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="12"
              viewBox="0 0 12 12"
              width="12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.76 3.2c.3.29.32.76.04 1.06l-4.25 4.5a.75.75 0 0 1-1.08.02L2.22 6.53a.75.75 0 0 1 1.06-1.06l1.7 1.7L8.7 3.24a.75.75 0 0 1 1.06-.04Z"
                fill="currentColor"
              />
            </svg>
          </span>
          textLayer3
        </div>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: calc(100% - 2rem);
  margin-top: 0.5rem;
  padding-left: 2rem;
}

.emotion-1 {
  width: calc(50% - 2.5rem);
  margin-right: 1.25rem;
}

.emotion-2 {
  -webkit-align-self: baseline;
  -ms-flex-item-align: baseline;
  align-self: baseline;
  width: calc(50% - 1.5rem);
  width: calc(50% - 0.5rem);
}

.emotion-3 {
  font-size: 0.813rem;
  color: black;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.emotion-4 {
  width: 1.5rem;
  margin-left: 0.25rem;
}

.emotion-5 {
  color: black;
  font-size: 0.813rem;
  width: 1.5rem;
  height: 1.5rem;
}

.emotion-5:hover {
  color: black;
}

<div>
    <div
      class="emotion-0"
    >
      <div
        class="ms-TextField emotion-1 root-110"
      >
        <div
          class="ms-TextField-wrapper"
        >
          <div
            class="ms-TextField-fieldGroup fieldGroup-111"
          >
            <input
              aria-invalid="false"
              class="ms-TextField-field field-112"
              id="TextField8"
              placeholder=""
              type="text"
              value="Propp"
            />
          </div>
        </div>
      </div>
      <div
        aria-owns="fluent-listbox2"
        class="fui-Dropdown emotion-2 ___88nkp20_1nwmw84 f1aa9q02 f16jpd5f f1jar5jt fyu767a f1ewtqcl f14t3ns0 f1exfvgq f10pi13n f14a1fxs f3e99gv fhljsf7 f1gw3sf2 f13zj6fq f1mdlcz9 f1a7op3 f1gboi2j f1cjjd47 ffyw7fx f1kp91vd f1ibwz09 f14pi962 f1lh990p f1jc6hxc f13evtba f1yk9hq fhwpy7i f14ee0xe f1xhbsuh fv8e3ye ftb5wc6 fjw5xc1 f1xdyd5c fatpbeo fb7uyps f1cmft4k f1x58t8o f1ibeo51 fxugw4r f192inf7 f5tn483 f1vxd6vx f1ojsxk5 fzkkow9 fcdblym fg706s2 fjik90z fj3muxo f1akhkt f1c1zstj f1lxtadh fvcxoqz f1ub3y4t flmw63s f1m52nbi fvs00aa f1assf6x fqhmt4z f4ruux4"
        style="border-radius: 2px; border: 1px solid #605e5c; font-size: 0.813rem; height: 1.5rem; line-height: 1.5rem;"
      >
        <button
          aria-activedescendant="fluent-option3"
          aria-expanded="true"
          class="fui-Dropdown__button ___1jkuyn2_1ro7mo8 f122n59 f1c21dwh fre7gi1 f1358rze fqdk4by f1rvrf73 f1ewtqcl fxc4j92 f14mj54c f1k6fduh f13qh94s fk6fouc f12nh0o2 f1869bpl f1o700av fly5x3f ftqa4ok fy9rknc figsok6 fwrc4pm f1khb0e9 fdw0yi8 f1jnq6q7 f1xile11"
          role="combobox"
          type="button"
        >
          <div
            class="emotion-3"
          >
            textLayer2, textLayer3
          </div>
          <span
            class="fui-Dropdown__expandIcon ___wu62wz0_l0el1i0 f1ewtqcl fxkbij4 ftgm304 f4ybsrx f12w6cgp f8bv1bt fvc9v3g f1h9en5y"
          >
            <svg
              aria-hidden="true"
              class="___12fm75w_v8ls9a0 f1w7gpdv fez10in fg4l7m0"
              fill="currentColor"
              height="1em"
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M15.85 7.65c.2.2.2.5 0 .7l-5.46 5.49a.55.55 0 0 1-.78 0L4.15 8.35a.5.5 0 1 1 .7-.7L10 12.8l5.15-5.16c.2-.2.5-.2.7 0Z"
                fill="currentColor"
              />
            </svg>
          </span>
        </button>
        <span
          hidden=""
        />
      </div>
      <div
        class="emotion-4"
      >
        <button
          aria-label="delete.property"
          class="ms-Button ms-Button--icon emotion-5 root-121"
          data-is-focusable="true"
          data-testid="delete-icon"
          title="delete.property"
          type="button"
        >
          <span
            class="ms-Button-flexContainer flexContainer-122"
            data-automationid="splitbuttonprimary"
          >
            <i
              aria-hidden="true"
              class="ms-Icon root-105 emotion-6 ms-Button-icon icon-124"
              data-icon-name="StatusCircleErrorX"
              style="font-family: \\"FabricMDL2Icons-9\\";"
            >
              
            </i>
          </span>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
